hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCommitPartRequest.java

```diff
@@ -229,7 +229,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       // AllocatedBlock. The space occupied by the Key shall be based on
       // the actual Key size, and the total Block size applied before should
       // be subtracted.
-      long correctedSpace = omKeyInfo.getDataSize() * factor -
+      long correctedSpace = omKeyInfo.getReplicatedSize() -
```

Comment:
This subtraction is using blockSize*factor, can we think what we need to consider here for EC keys?

Reply:
This calculation is correct for EC keys. When a block is allocated, it adds blockSize * factor to the bucket usage, assuming the block will be filled. Factor here is:

```
int factor = omKeyInfo.getReplicationConfig().getRequiredNodes();
```

So for EC 3-2 it will be 5, for 6-3 it will be 9 etc. If the block was exactly filled, this would be the actual space used by the EC block including replication.

When the block is completed, we calculate the actual space used, and then remove the pre-allocated amount, leave the correct valie of usage added to the bucket.